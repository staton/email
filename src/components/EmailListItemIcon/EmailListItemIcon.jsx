import React, { Component } from 'react';
import PropTypes from 'prop-types';

const propTypes = {
    content: PropTypes.any,
    isVisible: PropTypes.bool,
    onClick: PropTypes.func
};

const defaultProps = {
    content: null,
    isVisible: true,
    onClick: () => {}
};

class EmailListItemIcon extends Component {

    constructor() {
        super();

        this.handleClicked = this.handleClicked.bind(this);
    }

	render() {
		return (
            <div 
                className={this.getClassName()}
                onClick={this.handleClicked}
            >
                {(this.props.isVisible) ? this.props.content : null}
            </div>
		);
    }

    /**
     * Gets the class name(s) for this component.
     * @returns {string} The class name(s).
     */
    getClassName() {
        let className = 'EmailListItemIcon';

        if (this.props.isVisible) {
            className += ' EmailListItemIcon--visible';
        }

        return className;
    }

    /**
     * Called when the user clicks the email list item icon.
     * @param {object} e The event.
     */
    handleClicked(e) {
        e.preventDefault();
        e.stopPropagation();
        this.props.onClick();
    }

}

EmailListItemIcon.propTypes = propTypes;
EmailListItemIcon.defaultProps = defaultProps;

export default EmailListItemIcon;