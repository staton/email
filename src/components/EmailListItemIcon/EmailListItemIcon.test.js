import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow, mount } from 'enzyme';
import EmailListItemIcon from './EmailListItemIcon';

describe('<EmailListItemIcon />', () => {

    it('renders correctly', () => {
        const wrapper = shallow(
            <EmailListItemIcon
                content="X" 
                isVisible={true}
            />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('has the ..--visible class when visible', () => {
        const wrapper = shallow(
            <EmailListItemIcon
                content="X" 
                isVisible={true}
            />);
        expect(wrapper.find('.EmailListItemIcon--visible').length).toEqual(1);
    });

    it('does not have the ..--visible class when not visible', () => {
        const wrapper = shallow(
            <EmailListItemIcon
                content="X" 
                isVisible={false}
            />);
        expect(wrapper.find('.EmailListItemIcon--visible').length).toEqual(0);
    });

    it('handles click events', () => {
        const onClick = jest.fn();
        const wrapper = mount(
            <EmailListItemIcon
                content="X" 
                isVisible={true}
                onClick={onClick}
            />);
        const icon = wrapper.find('.EmailListItemIcon');
        icon.simulate('click');
        expect(onClick).toHaveBeenCalledTimes(1);
    });
    
});