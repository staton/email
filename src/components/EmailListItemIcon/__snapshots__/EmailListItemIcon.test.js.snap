// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EmailListItemIcon /> renders correctly 1`] = `
<div
  className="EmailListItemIcon EmailListItemIcon--visible"
  onClick={[Function]}
>
  X
</div>
`;
